pkg/cluster/fixssh.go

```diff
@@ -124,52 +153,122 @@ func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup,
+				updateSSHPool = false
+			}
+			if strings.EqualFold(*p.ID, ilbBackendPoolID) {
+				updateILBPool = false
+			}
 		}
 	}
 
 	if updateSSHPool {
-		m.log.Printf("Adding NIC %s to Internal Load Balancer SSH Backend Address Pool %s", nicName, sshID)
+		m.log.Printf("Adding NIC %s to Internal Load Balancer SSH Backend Address Pool %s", nicName, sshBackendPoolID)
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &sshID,
+			ID: &sshBackendPoolID,
 		})
 	}
 
 	if updateILBPool {
-		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbID)
+		m.log.Printf("Adding NIC %s to Internal Load Balancer API Address Pool %s", nicName, ilbBackendPoolID)
 		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-			ID: &ilbID,
+			ID: &ilbBackendPoolID,
 		})
 	}
 
 	return updateSSHPool || updateILBPool
 }
 
+func (m *manager) updateV1ELBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, resourceGroup string, infraID string) error {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		return fmt.Errorf("unable to update NIC as there are no IP configurations for %s", nicName)
+	}
+
+	var updateV1ELBPool bool
+	lb, err := m.loadBalancers.Get(ctx, resourceGroup, infraID, "")
+	elbBackendPoolID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, infraID)
+	lbpool := *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools
+	var index int
+	for i, p := range lbpool {
+		if strings.EqualFold(*p.ID, elbBackendPoolID) {
+			updateV1ELBPool = true
+			index = i
+		}
+	}
+
+	if updateV1ELBPool {
+		m.log.Printf("Removing NIC %s from Public Load Balancer API Address Pool %s", nicName, elbBackendPoolID)
+		lbpool = append(lbpool[:index], lbpool[index+1:]...)
+		(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &lbpool
+		m.log.Printf("Updating Network Interface %s", nicName)
+		err = m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, *nic)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func (m *manager) updateELBAddressPool(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		m.log.Warnf("unable to update NIC as there are no IP configurations for %s", nicName)
+		return false
+	}
+
+	var elbBackendPool string
+	switch m.doc.OpenShiftCluster.Properties.ArchitectureVersion {
+	case api.ArchitectureVersionV1:
+		elbBackendPool = infraID + "-public-lb-control-plane-v4"
+	case api.ArchitectureVersionV2:
+		elbBackendPool = infraID
+	}
```

Comment:
Because the switch case has only two options at this point and one of the options is just to use the infraID verbatim, you could simplify it into a single if statement to see if its version 1 and add the extra info.

Reply:
```
elbBackendPool := infraID
if m.doc.OpenShiftCluster.Properties.ArchitectureVersion == api.ArchitectureVersion {
  elbBackendPool = elbBackendPool + "-public-lb-control-plane-v4"
}
```

Reply:
```	elbBackendPool := infraID
	if m.doc.OpenShiftCluster.Properties.ArchitectureVersion == api.ArchitectureVersionV1 {
		elbBackendPool = elbBackendPool + "-public-lb-control-plane-v4"
	}